language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/FieldAccessCompletionResolver.java

```diff
@@ -82,6 +83,25 @@ public Optional<TypeSymbol> transform(SimpleNameReferenceNode node) {
         Optional<Symbol> symbol = this.getSymbolByName(context.visibleSymbols(context.getCursorPosition()),
                 node.name().text());
 
+        if (node.parent().kind() == SyntaxKind.FIELD_ACCESS) {
+            Optional<SemanticModel> semanticModel = this.context.currentSemanticModel();
+            if (semanticModel.isEmpty()) {
+                return Optional.empty();
+            }
+
+            Optional<TypeSymbol> typeSymbol = semanticModel.get()
+                    .typeOf(((FieldAccessExpressionNode) node.parent()).expression());
+            if (typeSymbol.isEmpty()) {
+                if (symbol.isEmpty()) {
+                    return Optional.empty();
+                }
+
+                return SymbolUtil.getTypeDescriptor(symbol.get());
+            }
+
+            return SymbolUtil.getTypeDescriptor(typeSymbol.get());
```

Comment:
This can be simplified as follows.
```suggestion
            if (typeSymbol.isPresent()) {
                   return SymbolUtil.getTypeDescriptor(typeSymbol.get());
            }
```